src/lib.rs

```diff
@@ -17,10 +17,32 @@ use deunicode::deunicode_char;
 /// assert_eq!(slugify("You & Me"), "you-me");
 /// assert_eq!(slugify("user@example.com"), "user-example-com");
 /// ```
+#[cfg(not(target_family = "wasm"))]
 pub fn slugify<S: AsRef<str>>(s: S) -> String {
     _slugify(s.as_ref())
 }
 
+/// Convert any unicode string to an ascii "slug" (useful for file names/url components)
+///
+/// The returned "slug" will consist of a-z, 0-9, and '-'. Furthermore, a slug will
+/// never contain more than one '-' in a row and will never start or end with '-'.
+///
+/// ```rust
+/// use self::slug::slugify;
+///
+/// assert_eq!(slugify("My Test String!!!1!1"), "my-test-string-1-1");
+/// assert_eq!(slugify("test\nit   now!"), "test-it-now");
+/// assert_eq!(slugify("  --test_-_cool"), "test-cool");
+/// assert_eq!(slugify("Æúű--cool?"), "aeuu-cool");
+/// assert_eq!(slugify("You & Me"), "you-me");
+/// assert_eq!(slugify("user@example.com"), "user-example-com");
+/// ```
+#[cfg(target_family = "wasm")]
+#[wasm_bindgen]
+pub fn slugify(s: String) -> String {
```

Comment:
This will break anyone using slugify within wasm as a dependency.

- It may be sufficient to just check if our target is `cdynlib`?
- Maybe create a second `#[doc(hidden)]` method and decorate it with `#[wasm_bindgen(js_name = slugify)]`?

I don't have enough experience here to know the right approach.

Reply:
How do you mean break it? If you're referring to the removal of the <S: AsRef<str>>, the problem was you can't wasm_bindgen to a function which takes arguments with lifetimes.

```shell
error: can't #[wasm_bindgen] functions with lifetime or type parameters
  --> src/lib.rs:42:15
   |
42 | pub fn slugify<S: AsRef<str>>(s: S) -> String {
   |               ^^^^^^^^^^^^^^^

error: could not compile `slug` due to previous error
```

I can still do your suggestion about renaming and hiding from docs though.

Reply:
> How do you mean break it?

It will break anyone who's using this as a dependency and compiling their top-level crate to wasm.

> Another possible way would be to create a feature that's called like wasm-bindings and use that as the conditions for the different functions instead the target family

That would work. But I'd also consider adding a new function:

```rust
#[wasm_bindgen(js_name = "slugify")]
#[doc(hidden)]
pub fn slugify_bindgen(s: String) -> String;
```

Reply:
Ah, ok, you've already changed it.